phoenix-core/src/main/java/org/apache/phoenix/iterate/OrderedResultIterator.java

```diff
@@ -130,9 +135,9 @@ public OrderedResultIterator(ResultIterator delegate, List<OrderByExpression> or
             SizedUtil.OBJECT_SIZE + estimatedRowSize;
 
         // Make sure we don't overflow Long, though this is really unlikely to happen.
-        assert(limit == null || Long.MAX_VALUE / estimatedEntrySize >= limit);
+        assert (limit == null || Long.MAX_VALUE / estimatedEntrySize >= limit + this.offset);
```

Comment:
Based on the above diff, you added a space after "assert". Please remove so our formatting is consistent.